op-interop-mon/README.md

```diff
@@ -0,0 +1,51 @@
+# Optimism Interop Monitor
+
+## Purpose
+The Optimism Interop Monitor is a service that monitors cross-chain interactions between OP Protocol chains to detect and report any invalid messages on chains. It helps ensure the reliability and correctness of cross-chain communication.
+
+Interop Monitor's primary output is a collection of metrics which can alert operators for fast response and insight.
+The design of this service follows the [Design Doc](https://github.com/ethereum-optimism/design-docs/pull/222),
+and boilerplate structure was pulled from the adjacent `op-dripper`.
+
+## Architecture
+The service consists of several key components working together:
+
+- A main service (`InteropMonitorService`) that coordinates everything
+- A set of RPC Clients specified from command line, and given to each sub-component
+- Multiple `Finder` instances that scan chains for relevant transactions
+- Multiple `Updater` instances that take jobs for their chain and update them
+- A `Maintainer` that processes and tracks the state of cross-chain messages
+- Metrics reporting for monitoring and alerting
+
+The components communicate through channels and maintain state about message status.
+
+## Maintainer
+The `Maintainer` is responsible for tracking the status of cross-chain messages over time. It:
+
+- Receives jobs from Finders by draining the Finders' outbox
+- Processes each job to determine its current status (valid/invalid/missing)
+- Maintains state about message status and updates
+- One large channel as a looping queue of recurring jobs
+- Emits metrics about message status
+- Jobs are dropped once no longer needed for monitoring
+
+### Updaters
+Updaters are chain specific processors that take jobs and update them. They can be used to batch requests and run in parallel.
+
+## Finders
+Finders scan individual chains for relevant cross-chain transactions. Each Finder:
+
+- Subscribes to new blocks on its assigned chain
```

Comment:
Can we clarify here and also simplify to?


> Finders scan individual chains for `ExecutingMessage` events from that chain's `CrossL2Inbox`.

Reply:
I'm seeing a fair bit of "cross chain transactions" terminology and I think it would help to use the initiating / executing message terminology.

In particular, it would be nice to more overtly associate jobs with executing messages, and then state that updaters are basically trying to find initiating messages for one or more jobs on a particular chain (if I have that right).

Reply:
Yes I totally agree. At the start of putting this together I was slightly more agnostic about what a job would be, but it is obviously a 1:1 mapping with an Executing Message.

I also agree that IM/EM terms are better than this "cross-chain transaction" phrasing. Internally they are called Executing Messages.